Content.Server/Chemistry/EntitySystems/ChemMasterSystem.cs

```diff
@@ -171,79 +168,164 @@ private void DiscardReagents(ChemMasterComponent chemMaster, string reagentId, F
 
         private void OnCreatePillsMessage(EntityUid uid, ChemMasterComponent chemMaster, ChemMasterCreatePillsMessage message)
         {
+            var user = message.Session.AttachedEntity;
+            var maybeContainer = _itemSlotsSystem.GetItem(chemMaster.Owner, SharedChemMaster.OutputSlotName);
+            if (maybeContainer is not { Valid: true } container
+                || !TryComp(container, out ServerStorageComponent? storage)
+                || storage.Storage is null)
+            {
+                return; // output can't fit pills
+            }
+            
+            // Ensure the number is valid.
+            if (message.Number == 0 || message.Number > storage.StorageCapacityMax - storage.StorageUsed) return;
+            
             // Ensure the amount is valid.
-            if (message.Amount == 0 || message.Amount > chemMaster.PillProductionLimit)
-                return;
+            if (message.Amount == 0 || message.Amount > chemMaster.PillDosageLimit) return;
+
+            var needed = message.Amount * message.Number;
+            if (!WithdrawFromBuffer(chemMaster, needed, user, out var withdrawal)) return;
+            
+            Label(container, message.Label);
+            
+            for (var i = 0; i < message.Number; i++)
+            {
+                var item = Spawn("Pill", Transform(container).Coordinates);
```

Comment:
probably should replace with a const in the system instead of it being embedded here